### PR TITLE
6 get article id/comments

### PR DESCRIPTION
**Pull request by nicomaz**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -104,7 +104,63 @@ describe("GET /api/articles/articles:id", () => {
...
+  it("200: responds with an array of comments of given article_id", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        const expectedComment = {
+          body: expect.any(String),
+          votes: expect.any(Number),
+          author: expect.any(String),
+          article_id: expect.any(Number),
```
Comment (popatre) [inline __tests__/app.test.js]: need to expect article id of 3

**Review by popatre — CHANGES_REQUESTED** [review]
Some good stuff here: two things

- [ ] Need to expect the article if you've requested the comments to be linked to
- [ ] Need a test: status 200 - serve an empty array when the article exists but has no comments

File: __tests__/app.test.js [hunk 2]
```diff
@@ -117,7 +117,6 @@ describe("GET /api/articles/:article_id/comments", () => {
           body: expect.any(String),
           votes: expect.any(Number),
           author: expect.any(String),
-          article_id: expect.any(Number),
```
Comment (ghost) [inline __tests__/app.test.js]: You could have just put the correct id here and your matcher in the forEach would have still worked.

File: controllers/comments.controllers.js [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
-      .then((comments) => {
-        res.status(200).send({ comments });
-      })
-      .catch(next);
-  };
\ No newline at end of file
+  const { article_id } = req.params;
+  const articlePromises = [selectCommentsByArticleId(article_id)];
+
+  if (article_id) {
```
Comment (ghost) [inline controllers/comments.controllers.js]: The article id here isn't optional, you could resolve all of this in a Promise.all()

**Review by ghost — APPROVED** [review]
Nice to see the empty comments test in. Good job, merge and move on.
